pkg/config/config.go

```diff
@@ -0,0 +1,114 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"fmt"
+	ioutil "io/ioutil"
+	"sync"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/config/v1alpha1"
+)
+
+// ControllerConfiguration defines the functions necessary to parse a config file
+// and to configure the Options struct for the ctrl.Manager
+type ControllerConfiguration interface {
```

Comment:
I think a better name would be `Controller_Manager_Configuration`, none of the settings are controller-level settings. Some comment applies for the types

Reply:
Snake case like that?

Reply:
No sorry, the underscores where just to highlight the difference

Reply:
Great, I'll get this updated. Thanks for the feedback :)

Reply:
All updated. Thanks again, @alvaroaleman